src/OrleansAzureUtils/Hosting/AzureSilo.cs

```diff
@@ -110,9 +110,16 @@ internal static ClusterConfiguration DefaultConfiguration(IServiceRuntimeWrapper
             {
                 config.Globals.DataConnectionString = serviceRuntimeWrapper.GetConfigurationSettingValue(AzureConstants.DataConnectionConfigurationSettingName);
             }
-            catch (RoleEnvironmentException)
+            catch (Exception exc)
             {
-                config.Globals.DataConnectionString = null;
+                if (exc.ToString().Contains("RoleEnvironmentException"))
```

Comment:
This is probably safe, but what we really want it to ensure the type, and ToString has more than that in it.
Suggets exc.GetType().Name.Contains(...)